src/App.tsx

```diff
@@ -10,17 +10,21 @@ import { Loader } from './components/Loader';
 import { Todo } from './types/Todo';
 import { getTodos } from './api';
 
+type PossibleCases = boolean | Todo[] | Todo | null | string;
+
 export const App: React.FC = () => {
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const [todos, setTodos] = useState<Todo[]>([]);
   const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
   const [titleFilter, setTitleFilter] = useState<string>('');
-  const [filter, setFilter] = useState<string>('');
+  const [filter, setFilter] = useState<PossibleCases>('');
```

Comment:
I think this is still an issue. 
Filter should be an `enum` with possible filtering options. You can create this `enum` in `types/Todo.ts` and then import to `App.tsx` and to `TodoFilter.tsx`.
```
export const enum Status {
  ALL = 'all',
  ACTIVE = 'active',
  COMPLETED = 'completed',
}
```

You can access enum just like that: `Status.ALL`

You can read more about enums here: https://www.typescriptlang.org/docs/handbook/enums.html
Also think of an advantage of using `const enum` rather than `enum` :)